src/tests/ie_test_utils/common_test_utils/snippets_test_utils.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets_test_utils.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "ngraph/pattern/op/label.hpp"
+
+namespace ngraph {
+namespace builder {
+namespace subgraph {
+
+using ngraph::snippets::op::Subgraph;
+
+SnippetsRestoreResultInputName::SnippetsRestoreResultInputName() {
```

Comment:
Why common? And why this namespace?>

Reply:
I agree, why this is common tests utils?

Reply:
This is in common test utils because it's used in [cpu unit test](https://github.com/IvanNovoselov/openvino/blob/4767b226bea75a7b367067b4191eed87c42d9db8/src/tests/unit/cpu/ngraph_transformations/snipptes_mark_skipped.cpp#L23) and [ie funstional test](https://github.com/IvanNovoselov/openvino/blob/78548c806b880a820ae398308089f04fce24f28f/src/tests/functional/inference_engine/snippets/src/pass/collapse_subgraph.cpp#L19).

Please, also note that this transformation is needed only because TransformationTestsF calls ngraph::pass::CheckUniqueNames, which expect different Result-related ops and tensor naming rules compared to CPU plugin. See the explanation and proposed solutions in the [snippets_test_utils.hpp](https://github.com/IvanNovoselov/openvino/blob/4767b226bea75a7b367067b4191eed87c42d9db8/src/tests/ie_test_utils/common_test_utils/snippets_test_utils.hpp#L13-L24). So we won't need this transformation at all, if we resolve the conflict between ngraph::pass::CheckUniqueNames and CPU plugin Result-related nodes naming rules.

Reply:
Another option is to include it inference_engine_snippets library. This transformation is rather small, so no significant binary size increase is expected. However, this pass is needed only for tests, so this is not the best solution.

Reply:
As discussed with @dmitry-gorokhov offline, I'm going to update Subgraph node naming policy, so the subgraph will keep the name of the last attached node. This change will eliminate the need in this transformation.

Summary: The transformation is removed.